opensrp-chw-core/src/test/java/org/smartregister/chw/core/interactor/CoreAncMemberProfileInteractorTest.java

```diff
@@ -0,0 +1,134 @@
+import org.mockito.MockitoAnnotations;
+import org.powermock.reflect.Whitebox;
+import org.robolectric.util.ReflectionHelpers;
+import org.smartregister.Context;
+import org.smartregister.CoreLibrary;
+import org.smartregister.chw.anc.AncLibrary;
+import org.smartregister.chw.anc.contract.BaseAncMemberProfileContract;
+import org.smartregister.chw.anc.domain.MemberObject;
+import org.smartregister.chw.anc.domain.Visit;
+import org.smartregister.chw.anc.repository.VisitRepository;
+import org.smartregister.chw.anc.util.AppExecutors;
+import org.smartregister.chw.core.BaseUnitTest;
+import org.smartregister.chw.core.application.CoreChwApplication;
+import org.smartregister.chw.core.contract.AncMemberProfileContract;
+import org.smartregister.chw.core.dao.AncDao;
+import org.smartregister.chw.core.repository.ChwTaskRepository;
+import org.smartregister.domain.Task;
+import org.smartregister.repository.TaskRepository;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Executors;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class CoreAncMemberProfileInteractorTest extends BaseUnitTest {
+
+    private final Visit lastVisit = new Visit();
+    private final Set<Task> taskList = new HashSet<>();
+    private CoreAncMemberProfileInteractor interactor;
+    private MemberObject memberObject;
+    @Mock
+    private BaseAncMemberProfileContract.InteractorCallBack interactorCallBack;
+    @Mock
+    private AncLibrary ancLibrary;
+    @Mock
+    private VisitRepository visitRepository;
+    @Mock
+    private CoreChwApplication chwApplication;
+    @Mock
+    private TaskRepository taskRepository;
+    @Mock
+    private ChwTaskRepository chwTaskRepository;
+    @Mock
+    private AncMemberProfileContract.InteractorCallBack callback;
+
+    private AppExecutors appExecutors = new AppExecutors(Executors.newSingleThreadExecutor(), Executors.newSingleThreadExecutor(), Executors.newSingleThreadExecutor());
+
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        Context context = Context.getInstance();
+        CoreLibrary.init(context);
+
+        //Auto login by default
+        context.session().start(context.session().lengthInMilliseconds());
+
+        interactor = Mockito.mock(CoreAncMemberProfileInteractor.class, Mockito.CALLS_REAL_METHODS);
+
+        memberObject = Mockito.mock(MemberObject.class);
+        memberObject.setPregnancyRiskLevel("Low");
+        memberObject.setBaseEntityId("some-base-entity-id");
+        memberObject.setFamilyName("Some Family Name");
+
+        ReflectionHelpers.setStaticField(AncLibrary.class, "instance", ancLibrary);
+    }
+
+    @Test
+    public void getPregnancyRiskDetailsTest() {
+        interactor.getPregnancyRiskDetails(memberObject);
+        Assert.assertNull(memberObject.getPregnancyRiskLevel());
```

Comment:
Not sure I understand this test. Shouldn't the test ascertain that you can get the correct pregnancy risk details? 🤷🏽‍♂️

Reply:
Let me look at this

Reply:
Actually this Methods returns the `memberObject`